bytecomp/bytegen.ml

```diff
@@ -1124,8 +1162,10 @@ let comp_function tc cont =
     | id :: rem -> Ident.add id pos (positions (pos + delta) delta rem) in
   let env =
     { ce_stack = positions arity (-1) tc.params;
```

Comment:
Note: the `positions` function here is `add_positions Ident.empty Fun.id`. I wonder if you could make `add_positions` a more global function to be able to use it here and reduce the total amount of code.